File: haskell/sum-of-multiples/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples, sumOfMultiplesDefault) where

sumOfMultiples :: [Int] -> Int -> Int
sumOfMultiples divisors n = sum multiples where
	multiples = [x | x <- [1..n-1], isMultiple x]
	isMultiple x = any (\y -> x `mod` y == 0) divisors

sumOfMultiplesDefault :: Int -> Int
sumOfMultiplesDefault n = sumOfMultiples [3, 5] n

File: typescript/sum-of-multiples/sum-of-multiples.ts
export default function SumOfMultiples(multiples: number[]) {
    return {
        to(max: number) {
            let sum = 0

            for (let i = 1; i < max; i += 1) {
                if (multiples.find((m) => i % m === 0)) {
                    sum += i
                }
            }

            return sum
        }
    }
}